servicesExternal/ipfs_service.py

```diff
@@ -23,21 +23,33 @@ def __init__(self, publicIpfs: bool=True):
         self.ipfs = ipfsapi.connect(self.ipfsConfig["host"], self.ipfsConfig["port"], self.ipfsConfig["protocol"])
 
     async def add_file(self, data: str):
-        if data == None or data == "":
-            raise ValueError("")
+        if data is None or data == "":
+            raise ValueError("Data is None or invalid.")
         
         data_parsed = json.loads(data)
         result = await self.ipfs.add(json.dumps(data_parsed, separators=(',',':')))
 
         return result["hash"]
 
     async def get_file(self, hash: str):
-        if hash == None or hash == "":
-            raise ValueError("")
+        if hash is None or hash == "":
+            raise ValueError("Hash is None or invalid.")
         
         try:
             result = await self.ipfs.cat(hash)
 
             return result
+        except ipfsapi.exceptions.ConnectionError as err:
+            print("IPFS connection error: ", err)
```

Comment:
All of these are great! But don't `print` just reraise the error with additional details if possible.